.github/workflows/udapte-deps.yml

```diff
@@ -0,0 +1,56 @@
+name: 'Update Nix Flake Inputs'
+on:
+  push:
+    branches:
+      - '_update-deps_runtimeverification_haskell-backend'
+      - '_update-deps_runtimeverification_llvm-backend'
+  workflow_dispatch:
+# Stop in progress workflows on the same branch and same workflow to use latest committed code
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  nix-flake-submodule-sync:
+    name: 'Nix flake submodule sync'
+    runs-on: ubuntu-latest
+    steps:
+      - id: config
+        run: |
+          ref=${{ github.ref }}
+          if [ "${{ github.event_name }}" == 'pull_request' ]; then
+            ref="${{ github.event.pull_request.head.sha }}"
+          fi
```

Comment:
This block should never fire now, and can be removed.

Also, we are using `ref=${{ github.ref }}` in this bash block, when we could probably just use `${GITHUB_HEAD_REF}` variable directly with is set by the github workflow?

Either way, this `if` statement doeeesn't make sense any more in a branch push, but it can be removed in follow-up.

Reply:
I'll make these changes in my local branch and make a PR with updates.